File: src-tauri/src/commands/connection.rs
use tauri::State;
use crate::db::{self, DbState};

#[tauri::command]
pub async fn connect_db(
    state: State<'_, DbState>,
    host: String,
    port: u16,
    user: String,
    password: String,
    database: String,
) -> Result<(), String> {
    let pool = db::connect(&host, port, &user, &password, &database).await?;

    let mut db = state.pool.lock().await;
    *db = Some(pool);

    log::info!("Connected to MySQL {}@{}:{}/{}", user, host, port, database);
    Ok(())
}

#[tauri::command]
pub async fn disconnect_db(state: State<'_, DbState>) -> Result<(), String> {
    let mut db = state.pool.lock().await;
    if let Some(pool) = db.take() {
        pool.close().await;
        log::info!("Disconnected from MySQL");
    }
    Ok(())
}

File: server/src/handlers/connection.ts
import { connect, disconnect } from '../db.js'

export async function connect_db(params: {
  host: string, port: number, user: string, password: string, database: string
}): Promise<null> {
  await connect(params.host, params.port, params.user, params.password, params.database)
  return null
}

export async function disconnect_db(_params: Record<string, never>): Promise<null> {
  await disconnect()
  return null
}
